mmdet/models/necks/bifpn.py

```diff
@@ -0,0 +1,327 @@
+# Based off of github.com/rwightman/efficientdet-pytorch
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import xavier_init
+from mmdet.core import auto_fp16
+
+from ..builder import NECKS
+
+from mmcv.cnn import normal_init
+
+act_fn_list = ["silu", "swish", "hswish", "relu", "relu6", "mish", "srelu"]
+
+class ActLayer(nn.Module):
+    def __init__(self, act_name):
+        super(ActLayer, self).__init__()
+        assert act_name in act_fn_list
+        self.act_fn = act_name
+
+    def forward(self, nodes):
+        # Activation function
+        if (self.act_fn == "silu"):
+            nodes = nodes * torch.sigmoid(nodes)
+
+        # # Quantization-friendly hard swish
+        elif (self.act_fn == "swish"):
+            nodes = nodes * F.relu6(nodes + numpy.float32(3)) * (1. / 6.)
+            nodes = nodes * torch.sigmoid(nodes)
+
+        elif (self.act_fn == "hswish"):
+            nodes = nodes * F.relu6(nodes + 3) / 6
+
+        elif (self.act_fn == "relu"):
+            nodes = F.relu(nodes)
+
+        elif (self.act_fn == "relu6"):
+            nodes = F.relu6(nodes)
+
+        elif (self.act_fn == "mish"):
+            nodes = nodes * F.tanh(F.softplus(nodes))
+
+        elif (self.act_fn == "srelu"):
+            beta = numpy.array([20.0])
+            beta = torch.autograd.Variable(torch.from_numpy(beta)) ** 2
+            beta = (beta ** 2).type(nodes.type())
+            safe_log = torch.log(torch.where(nodes > 0., beta * nodes + 1., torch.ones_like(nodes)))
+            nodes = torch.where((nodes > 0.), nodes - (1. / beta) * safe_log, torch.zeros_like(nodes))
+
+        return nodes
+
+
+class ConvBn(nn.Module):
```

Comment:
I believe that mmcv already has a lot of basic building blocks, including some of the ones introduced here. It'd be better to reuse those, though this is not a showstopper at the moment.

Reply:
I tried using ConvModule from mmcv to build ConvBn and SeparableConv2d, ended up taking up more GPU memory and lowered the batch size for the model. If there's another building block that mmcv has that may take less memory, please let me know

Reply:
I tried using ConvModule, it seems whatever I did previously was just an error, training can be done as normal. Issue now is that replacing the layers in BiFPN will lead to needing to retrain the model, as the checkpoint won't recognize the used layers. Let me know if the custom layers are fine, or if they have to be replaced

Reply:
I'd prefer to use as much standard mmcv's building blocks (including convolutions and activations) as possible.

Reply:
My suggestion is to put import of `DepthwiseSeparableConvModule` to `try` section and opt to use your custom building block only if the default one is not available in used version of mmcv.

Reply:
Please ignore my previous comment. It may introduce some inconsistency and incompatibility between different mmcv package versions. As long as we have fixed version of mmcv in requirements, let's leave your code as is.